src/Compilers/CSharp/Test/Emit3/Attributes/AttributeTests_Synthesized.cs

```diff
@@ -1816,6 +1816,45 @@ public static async void F()
                 Assert.Empty(asyncMethod.GetAttributes());
             }
         }
+
+        [Theory]
+        [MemberData(nameof(OptimizationLevelTheoryData))]
+        public void AsyncStateMachineAttribute_RuntimeAsync(OptimizationLevel optimizationLevel)
+        {
+            string source = """
+                using System.Threading.Tasks;
+
+                class Test
+                {
+                    public static async Task F()
+                    {
+                        await Task.Delay(0);
+                    }
+                }
+                """;
+
+            var options = TestOptions.CreateTestOptions(OutputKind.DynamicallyLinkedLibrary, optimizationLevel)
+                .WithMetadataImportOptions(MetadataImportOptions.All);
+
+            var compilation = CreateRuntimeAsyncCompilation(source, options);
+            var verifier = CompileAndVerify(compilation, verify: Verification.Skipped, symbolValidator: verify);
+            verifier.VerifyDiagnostics();
+
+            void verify(ModuleSymbol module)
+            {
+                var type = module.GlobalNamespace.GetMember<NamedTypeSymbol>("Test");
+                var asyncMethod = type.GetMember<MethodSymbol>("F");
+
+                var attributes = asyncMethod.GetAttributes();
+
+                // When runtime async is enabled, no state machine is generated,
+                // so there should be no AsyncStateMachineAttribute and no DebuggerStepThroughAttribute
+                Assert.Empty(attributes);
+
+                // Verify no state machine type was generated
+                Assert.Empty(type.GetTypeMembers("<F>d__0"));
```

Comment:
>"<F>d__0" [](http://example.com/codeflow?start=49&length=9)

I have no idea whether this name is the right name to use. I assume we don't expect to have any nested types. Let's assert that instead.